databroker/v1.py

```diff
@@ -1640,14 +1640,52 @@ def _from_v0_config(config, auto_register, name):
         dotted_object = '.'.join((contents['module'], contents['class']))
         handler_registry[spec] = dotted_object
     root_map = config.get('root_map')
+
+    transforms = _load_transforms(config.get('transforms'))
+
     return BlueskyMongoCatalog(metadatastore_db, asset_registry_db,
                                handler_registry=handler_registry,
                                root_map=root_map,
-                               name=name)
+                               name=name,
+                               transforms=transforms)
+
 
 _mongo_clients = {}  # cache of pymongo.MongoClient instances
 
 
+def _load_transforms(transforms):
+     """
+     Parse mapping of spec name to 'import path' into mapping to class itself.
+
+     Parameters
+     ----------
+     transforms : dict or None
+         Values may be string 'import paths' to classes or actual classes.
+
+     Examples
+     --------
+     Pass in name; get back actual class.
+
+     >>> _load_transforms({'descriptor': 'package.module.ClassName'})
+     {'descriptor': <package.module.ClassName>}
+
+     """
+     result = {}
+
+     if 'event' in transforms.keys() or 'datum' in transforms.keys():
+        raise NotImplementedError("Transforms for events and datum are not supported.")
+
+     for name in ('start', 'stop', 'resource', 'descriptor'):
+         transform = transforms.get(name)
+         if isinstance(transform, str):
+            module_name, _, class_name = transform.rpartition('.')
+            function = getattr(importlib.import_module(module_name), class_name)
+         else:
```

Comment:
The `parse_handler_registry` function accepts strings (such as we'd get if this came from a YAML config) or the classes themselves. This is different and I think could be potentially very surprising. It accepts strings, but if the user provides a class (or anything other than a string) it silently ignores the user's input and just assigns a pass-through function.

Reply:
Thanks for the feedback! Updated

Reply:
Thanks for the update. Looks closer, but it still differs from `parse_handler_registry` in that `parse_handler_registry` accepts the actual class, as in `{'AD_HDF5': MyHandler}`, which allows users to experiment with one-off classes that aren't importable from a module. I think the same thing could be useful for transformers, so it would be good to support `{'start': my_transformer}` as well as `{'start': 'my.packaged.transfomer'}`.

Reply:
But load_transforms is for loading from a config_file, and you can't have a class instance in the config file. I don't think calling it directly would be useful, because you would have to pass the result to the BlueskyMongoCatalog init, and you can just pass {'start': my_transformer} directly to the init.

Reply:
updated